doc/static-route/SONiC_static_route_bfd_hld.md

```diff
@@ -0,0 +1,300 @@
+# Scope
+
+This document describes the design of using BFD to detect static route nexthop and based on BFD session state, installing, updating, and removing static route to the system.<br  /> 
+
+One important assumption is that the BFD session (for a static route nexthop) in this BFD Static Route application is not shared with any other applications (i.e., creating or deleting static route BFD session has no impact on other applications).<br /> 
+
+<span style="color:blue"> <br>
+*Note: This document describes how to create local BFD session for a static route based on static route configuration. Creating BFD session in its peer system is out of the scope of this document.*<br />  
+</span>
+
+
+# Static Route BFD Design
+## Functional Requirements
+
+1. Create BFD session for each static route nexthop based on static route configuration <br>
+2. Delete BFD session if the nexthop is not needed (triggered by removing static route from configration)<br>
+3. Install, update, removing static route (i.e., update nexthop, remove static route if no nexthop) based on BFD session state change.<br>
+4. When the application (static route BFD) restarts, recover static routes and BFD session states from redis database (config_db, appl_db and state_db) without impacting any existing installed static routes and BFD sessions. <br  />  
+
+## System Overview with Static Route BFD
+A new component, StaticRouteBfd is introduced to support static route BFD. <br>
+
+In the following diagram, StaticRouteBfd monitors config_db STATIC_ROUTE_TABLE. If there is a static route configured with bfd, StaticRouteBfd creates BFD session by updating appl_db BFD_SESSION_TABLE. StaticRouteBfd also monitors state db for BFD session state change. Based on BFD session state, StaticRouteBfd updates appl_db STATIC_ROUTE_TABLE to add/delete static route or update its nexthop list.<br>
+
+To work with existing bgpcfgd StaticRouteMgr, an optional field "bfd" is introduced in STATIC_ROUTE_TABLE. When the "bfd" field present and is "true" in config db, StaticRouteMgr ignores this static route and StaticRouteBfd handles it. StaticRouteBfd writes this static route to appl_db based on BFD session state for this static route nexthop. <br>
+
+To avoid StaticRouteTimer deleting StaticRouteBfd created static route entry in appl_db, a new field "expiry" is introduced in appl_db STATIC_ROUTE_TABLE schema. StaticRouteBfd sets "expiry"="false" when it writes a static route to appl_db STATIC_ROUTE_TABLE. When StaticRouteTimer see "expiry"="false", it ignores this static route entry. <br>
+
+
+<img src="static_rt_bfd_overview.png" width="500">
```

Comment:
Can I assume red color coded are newly introduced modules/interactions? if no, please color code accordingly.

Reply:
![image](https://user-images.githubusercontent.com/34145258/229638420-0b55503c-4d08-4b6e-8075-82689db4b25a.png)
I think, only persistent static routes are pushed to config-db and this route will be configured to FRR and handled back as regular route via fpmsyncd for HW programming. staticroutebfd handles only the persistent static route case? how about non-persistent static route?

Reply:
Suggest to add the details for SAI BFD offload APIs being used for this feature.

Reply:
updated the diagram: bfd and expiry are newly added fields. the interactions are color coded (see color code at top-right corner). fixed typos. Thanks.

Reply:
The target of this design is to handle static routes entry from config_db only. non-persistent static route case is a more complicated scenario, because that StaticRouteBfd also need to write static route to AppDB to let StaticRouteMgr (appl_db) to install the route to FRR.